scripts/repo-variable-writer.go

```diff
@@ -0,0 +1,373 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log/slog"
+	"net/http"
+	"os"
+	"strings"
+	"sync"
+	"time"
+)
+
+type Project struct {
+	Name          string   `json:"name"`
+	Organization  string   `json:"organization"`
+	SubComponents []string `json:"sub_components"`
+}
+
+type Projects struct {
+	Projects []Project `json:"projects"`
+}
+
+const (
+	LocProjMeta         string = "../projects/project.json"
+	EnvGithubPatVarName string = "GH_TOKEN"
+)
+
+func toPtr[T any](v T) *T {
+	return &v
+}
+
+func handleHTTPCall(method string, url string, timeOut time.Duration,
+	body io.Reader, headers map[string]string) (*http.Response, error) {
+
+	client := &http.Client{
+		Timeout: timeOut,
+	}
+
+	req, err := http.NewRequest(method, url, body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create newRequest, method: %s, url: %s, Reason: %v", method, url, err)
+	}
+
+	for k, v := range headers {
+		req.Header.Add(k, v)
+	}
+
+	v, err := client.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("failed in client http request, method: %s, url: %s, Reason: %v", method, url, err)
+	}
+	return v, nil
+}
+
+func loadProjectMetadata(locProj string) (resp *Projects, err error) {
+	_b, err := os.ReadFile(locProj)
+	if err != nil {
+		return
+	}
+
+	err = json.Unmarshal(_b, &resp)
+	return
+}
+
+func fetchLatestRelease(org, proj string) (*string, error) {
+	url := fmt.Sprintf(
+		"https://api.github.com/repos/%s/%s/releases/latest", org, proj)
```

Comment:
Maybe there is a way to make this "fetch release" which also has the capability to fetch the latest release. Giving functions more depth and avoid shallow functions

Reply:
not sure what you meant my shallow function in this context